core/templates/dev/head/I18nFooter.ts

```diff
@@ -20,37 +20,47 @@
-  function($http, $scope, $timeout, $translate, UserService) {
-    // Changes the language of the translations.
-    var preferencesDataUrl = '/preferenceshandler/data';
-    var siteLanguageUrl = '/save_site_language';
-    $scope.supportedSiteLanguages = constants.SUPPORTED_SITE_LANGUAGES;
+angular.module('oppia').directive('i18nFooter', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/i18n-footer.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        '$http', '$timeout', '$translate', 'UserService',
+        function($http, $timeout, $translate, UserService) {
+          var ctrl = this;
+          // Changes the language of the translations.
+          var preferencesDataUrl = '/preferenceshandler/data';
+          var siteLanguageUrl = '/save_site_language';
+          ctrl.supportedSiteLanguages = constants.SUPPORTED_SITE_LANGUAGES;
 
-    // The $timeout seems to be necessary for the dropdown to show anything
-    // at the outset, if the default language is not English.
-    $timeout(function() {
-      // $translate.use() returns undefined until the language file is fully
-      // loaded, which causes a blank field in the dropdown, hence we use
-      // $translate.proposedLanguage() as suggested in
-      // http://stackoverflow.com/a/28903658
-      $scope.currentLanguageCode = $translate.use() ||
-        $translate.proposedLanguage();
-    }, 50);
+          // The $timeout seems to be necessary for the dropdown to show
+          // anything at the outset, if the default language is not English.
+          $timeout(function() {
+            // $translate.use() returns undefined until the language file is
+            // fully loaded, which causes a blank field in the dropdown, hence
+            // we use $translate.proposedLanguage() as suggested in
+            // http://stackoverflow.com/a/28903658
+            ctrl.currentLanguageCode = $translate.use() ||
+              $translate.proposedLanguage();
+          }, 50);
 
-    $scope.changeLanguage = function() {
-      $translate.use($scope.currentLanguageCode);
-      UserService.getUserInfoAsync().then(function(userInfo) {
-        if (userInfo.isLoggedIn()) {
-          $http.put(siteLanguageUrl, {
-            site_language_code: $scope.currentLanguageCode
-          });
-        }
-      });
-    };
-  }
-]);
+          ctrl.changeLanguage = function() {
+            $translate.use(ctrl.currentLanguageCode);
+            UserService.getUserInfoAsync().then(function(userInfo) {
+              if (userInfo.isLoggedIn()) {
+                $http.put(siteLanguageUrl, {
+                  site_language_code: ctrl.currentLanguageCode
+                });
+              }
+            });
+          };
+        }]};
```

Comment:
The ']' should move to the next line to match the opening indentation.

The '}' should move to the line after that.

In general, when you see a gap that's too big between indentation levels, please fix it.

Reply:
Addressed in #7066.